### PR TITLE
implemented Diagram API

### PR DESCRIPTION
**Pull request by SihanChen46**

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -0,0 +1,234 @@
+aiofiles==22.1.0
```
Comment (esmeluo) [inline requirements.txt:1]: this seems like a lot of dependencies, are all of these needed for this application?

File: message_handler/message_handler.py [hunk 2]
```diff
@@ -1,3 +0,0 @@
-class MessageHandler:
```
Comment (esmeluo) [inline message_handler/message_handler.py:1]: do we still need the message_handler folder?

File: components/handlers/diagram_handler.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        )
+
+        chat_history = str(conversation_memory.chat_memory.messages)
+
+        chat_summary = SummaryChain().run(chat_history)
+        design_content = DesignChain().run(chat_summary)
+        diagram_code = DiagramChain().run(design_content)
+
+        resp_diagram = {
+            "diagramId": str(uuid.uuid4()),
```
Comment (esmeluo) [inline components/handlers/diagram_handler.py:23]: what is the purpose of diagramId? what is conversationId hard coded?

File: components/handlers/diagram_handler.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+            "conversationId": "xxx",
+            "diagramCode": diagram_code,
+            "status": "DONE",
```
Comment (esmeluo) [inline components/handlers/diagram_handler.py:26]: status is a string not a enum?

File: requirements.txt (hunk 1 above)
Comment (SihanChen46) [reply requirements.txt:1]: yes

File: components/handlers/diagram_handler.py (hunk 3 above)
Comment (SihanChen46) [reply components/handlers/diagram_handler.py:23]: I just think this is something for sure we'll need in the future, useless now.
my bad, conversationId should be from the clickDiagram

File: components/handlers/diagram_handler.py (hunk 4 above)
Comment (SihanChen46) [reply components/handlers/diagram_handler.py:26]: good catch

File: components/chains/summary_chain.py [hunk 5]
```diff
@@ -17,16 +17,16 @@ def output_keys(self) -> List[str]:
...
-        chain = self._build_conversation_chain()
+        chain = self._build_chain()
         outputs = {"response": chain.run(inputs)}
         return outputs
 
-    def _build_conversation_chain(self):
-        llm = ChatOpenAI(
+    def _build_chain(self):
+        llm = OpenAI(
+            model_name="text-davinci-003",
```
Comment (esmeluo) [inline components/chains/summary_chain.py:26]: why is this a different model?

**Pull request merged**